python/tvm/topi/sparse_fill_empty_rows.py

```diff
@@ -0,0 +1,117 @@
+
+# pylint: disable=no-else-return, too-many-locals, too-many-arguments, too-many-branches
+# pylint: disable=undefined-variable, invalid-name
+"""SparseFillEmptyRows operator"""
+from ..te import hybrid
+
+
+@hybrid.script
+def _sparse_fill_empty_rows(
+    sparse_indices,
+    sparse_values,
+    dense_shape,
+    default_value,
+    new_sparse_indices_shape,
+    new_sparse_values_shape,
+    empty_row_indicator_shape,
+):
+    default_value_ = int64(default_value[0])
+    new_sparse_indices = output_tensor(new_sparse_indices_shape, "int64")
+    new_sparse_values = output_tensor(new_sparse_values_shape, "int64")
+    empty_row_indicator = output_tensor(empty_row_indicator_shape, "int64")
+    idx = 0
+
+    if int64(sparse_indices.shape[0]) == int64(0):
+        for i in range(0, int64(new_sparse_indices_shape[0])):
+            new_sparse_indices[i, 0] = int64(i)
+            new_sparse_values[i] = default_value_
+            empty_row_indicator[i] = int64(1)
+            for k in range(1, int64(new_sparse_indices_shape[1])):
+                new_sparse_indices[i, k] = int64(0)
+
+        return (new_sparse_indices, new_sparse_values, empty_row_indicator)
+
+    else:
+        for i in range(0, int64(sparse_indices[0, 0])):
+            new_sparse_indices[idx, 0] = int64(i)
+            for k in range(1, int64(new_sparse_indices_shape[1])):
+                new_sparse_indices[idx, k] = int64(0)
+
+            new_sparse_values[idx] = default_value_
+            empty_row_indicator[i] = int64(1)
+            idx += 1
+
+        for i in range(0, int64(sparse_indices.shape[0])):
+            index = int64(sparse_indices[i, 0])
+            if i == 0:
+                new_sparse_indices[idx, 0] = index
+                for k in range(1, int64(new_sparse_indices_shape[1])):
+                    new_sparse_indices[idx, k] = int64(sparse_indices[i, k])
+                new_sparse_values[idx] = int64(sparse_values[i])
+                empty_row_indicator[index] = int64(0)
+                idx += 1
+            else:
+                prev_index = int64(sparse_indices[i - 1, 0] + 1)
+                for j in range(prev_index, index):
+                    new_sparse_indices[idx, 0] = int64(j)
+                    for k in range(1, int64(new_sparse_indices_shape[1])):
+                        new_sparse_indices[idx, k] = int64(0)
+                    empty_row_indicator[prev_index] = int64(1)
+                    new_sparse_values[idx] = default_value_
+                    idx += 1
+
+                new_sparse_indices[idx, 0] = index
+                for k in range(1, int64(new_sparse_indices_shape[1])):
+                    new_sparse_indices[idx, k] = int64(sparse_indices[i, k])
+                new_sparse_values[idx] = int64(sparse_values[i])
+                empty_row_indicator[index] = int64(0)
+                idx += 1
+
+        for i in range(
+            int64(sparse_indices[sparse_indices.shape[0] - 1, 0] + 1), int64(dense_shape[0])
```

Comment:
Why is there a `+1` in this expression? If the sparse indices have size 1 and the dense shape is size 2, then we wouldn't add any elements.

Reply:
Because we must start filling the next row up until dense_shape[0] - 1, 
For ex: 
`Sparse_indices element -> [[0]]` and dense_shape element is `[2] `
Then we have to fill in row "1" to make the final answer
`new_sparse_indices -> [[0],[1]]  `

Reply:
I have also added comments to shape func and this file so that should help. Let me know :)

Reply:
Sorry, I got it in my head that we were in CSR format. This looks correct.